uniswap-v2/logics/helpers/transfer_helper.rs

```diff
@@ -34,10 +35,18 @@ pub fn safe_transfer_from(
 
 #[inline]
 pub fn wrap(wnative: &AccountId, value: Balance) -> Result<(), PSP22Error> {
-    WnativeRef::deposit_builder(wnative)
+    match WnativeRef::deposit_builder(wnative)
         .transferred_value(value)
         .fire()
-        .unwrap()
+    {
+        Ok(res) => {
+            match res {
+                Ok(_) => Ok(()),
+                Err(err) => Err(err),
+            }
+        }
+        Err(_) => Err(PSP22Error::Custom(String::from("deposit failed"))),
```

Comment:
Rather than just saying deposit failure, think better to wrap ink error with PSP22Error, like 
```suggestion
        Err(err) => Err(PSP22Error::InkError(err)),
```

Reply:
I don't get it.
There is no InkError variant in PSP22Error enum

Reply:
Ah, yes, I forgot PSP22Error comes from OB. Yes, we cannot extend PSP22Error by ourselves. What I meant was it would be better to give better error message for ink side error.
But, as OB does not emit much information when it comes to ink failure, think "deposit failed" might be sufficient.